torch/nn/utils/rnn.py

```diff
@@ -310,12 +318,12 @@ def pad_packed_sequence(sequence, batch_first=False, padding_value=0.0, total_le
                              "total_length={} and max sequence length being {}"
                              .format(total_length, max_seq_length))
         max_seq_length = total_length
-    padded_output, lengths = torch._C._VariableFunctions._pad_packed_sequence(
+    padded_output, lengths = _VF._pad_packed_sequence(
         sequence.data, sequence.batch_sizes, batch_first, padding_value, max_seq_length)
-    if sequence.unsorted_indices is not None:
+    unsorted_indices = sequence.unsorted_indices
+    if unsorted_indices is not None:
         batch_dim = 0 if batch_first else 1
-        return padded_output.index_select(batch_dim, sequence.unsorted_indices), \
-            lengths[sequence.unsorted_indices]
+        return padded_output.index_select(batch_dim, unsorted_indices), lengths[unsorted_indices]
```

Comment:
Why did you have to rewrite this code? Because of optional unwrapping?

Reply:
Yes, our type refinement now is limited to TK_VAR only, we will need to expand it to a general case, see https://github.com/pytorch/pytorch/issues/23049